worker/worker.go

```diff
@@ -792,16 +842,69 @@ func joinErrors(errs []error) error {
 	}
 }
 
+func (w *worker) hostPriceTableByContract(ctx context.Context, hk consensus.PublicKey, hostIP string, revision *types.FileContractRevision) (pt rhp.HostPriceTable, err error) {
+	// TODO: Check if valid price table exists and handle the following edge cases.
```

Comment:
I'd appreciate some input on this. I think this is going to bite us during times of heavy uploads/migrations. We need some way to lock a contract across renter-host protocols for things that require revising a contract.